magento/pwa-studio
# [PWA-2312] Update CategoryTree.id GraphQL field to uid

## Description

- Updates `peregrine/lib/talons/MegaMenu/megaMenu.gql.js` to use the `uid` field instead of `id`
  - Updates relative tests and snapshots to match new field
- Removes `id` leftovers from
  - `peregrine/lib/talons/Breadcrumbs/breadcrumbs.gql.js`
  - `peregrine/lib/talons/CategoryList/categoryList.gql.js`
  - `peregrine/lib/talons/CategoryTree/categoryTree.gql.js`
  - `peregrine/lib/talons/RootComponents/Category/category.gql.js`
  - `peregrine/lib/talons/RootComponents/Category/categoryContent.gql.js`
    - It was need to disable the ESLint by unsing an inline comment. Hopefully we'll remove it as soon as support for multiple ids is implemented

## Related Issue

[PWA-2312](https://jira.corp.magento.com/browse/PWA-2312)

## Acceptance

### Verification Stakeholders

### Specification

## Verification Steps

- Open developer console as the page loads
- Look for megaMenu queries
- Ensure that the query now calls with `uid` instead of `id`

#### Test scenario(s) for direct fix/feature

#### Test scenario(s) for any existing impacted features/areas

#### Test scenario(s) for any Magento Backend Supported Configurations

#### Is Browser/Device testing needed?

#### Any ad-hoc/edge case scenarios that need to be considered?

## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)
## Checklist
